### ISSUE TITLE
Error: Note was not preceded by a highlight

### ISSUE THREAD

**Issue by EhuddR** [hadynz/obsidian-kindle-plugin#16]
Hello, the plugin does not work with MyClippings.txt.
Attached image of the error in the console

![image](https://user-images.githubusercontent.com/78237982/116827489-c6789980-ab5e-11eb-845c-621a7c0dc421.png)

**Comment by hadynz** [comment]
This is a very similar issue as https://github.com/hadynz/obsidian-kindle-plugin/issues/15. Please see [my comment][1] on if you would be open to sharing your `My Clippings.txt` file.

What's slightly different about your use case, is that the error you've encountered was no about being unable to parse highlight data. But not being able to associated a personal "note" that you've written against the right highlight.

I can make a fix to the plugin so it is not as "strict" in enforcing such an assumption, so the rest of your file is parsed. I'll let you know as soon as I make it.

[1]: https://github.com/hadynz/obsidian-kindle-plugin/issues/15#issuecomment-831168580

**Comment by hadynz** [comment]
Whilst it is not clear to me why you ran into this error (looking at your my clippings file will help me in doing so), I made an. update to the plugin to ignore the error that you just encountered in [0.1.1][1]. This means that the rest of the highlights in your file will be parsed.

Give it a go and let me know how it goes.

[1]: https://github.com/hadynz/obsidian-kindle-plugin/releases/tag/0.1.1

**Comment by EhuddR** [comment]
I just sent the My clippings file to your email for your analysis. I will try with the update you mention, Thank you very much.

**Comment by EhuddR** [comment]
I have tested the new update, however, the plugin does not pick up all the underlines in my notes leaving some incomplete. Here is a screenshot of the console.

![image](https://user-images.githubusercontent.com/78237982/116945919-6e18c900-ac3e-11eb-8fb2-bed34e66c55f.png)

I would like to know, if there is a possibility for the plugin to pick up the date information when an underline or annotation was made. I understand that this can be achieved using Nunjuck but I am not proficient in Java Script so it is a challenge for me.

**Comment by hadynz** [comment]
Thanks for emailing me your `My Clippings.txt` file via email.

> the plugin does not pick up all the underlines in my notes leaving some incomplete.
Can you give me examples of highlights (underlines) in your file that are missing?

**Comment by EhuddR** [comment]
For example: In the file My Clippings, there is a highlight named "BBC "which is a highlight on a newspaper made whit the calibre program and sended to my kindle. It seems that maybe is an error whit that kind of books. the same goes for "Revista Crisis" or "Rebellion"

**Comment by hadynz** [comment]
I just tested your file, and I can see all three books loaded fine. Did you mean that there are highlights within those books that are missing?

<img src="https://user-images.githubusercontent.com/315585/116982487-f53c6e80-ad1c-11eb-9099-e85304f27d02.png" width="400">

**Comment by EhuddR** [comment]
yes, the files are created but the plugin only collects a little underlining, not all the ones performed. that is, the texts are incomplete

**Comment by hadynz** [comment]
It is possible that the highlights you are expecting are missing from your My Clippings file in the first place. 

Do you mind if you give me examples of highlights that are inside My Clippings but not extracted by the plug-in?

**Comment by EhuddR** [comment]
Of corse. 

This image represents the Highlights retrived by the pugin.
![Obsidian_YbZfNSXiFc](https://user-images.githubusercontent.com/78237982/117063528-94943e00-acea-11eb-9e04-8d2382c186ec.png)
An this is all the highlites that I made on my kindle. This was imported manually.
![YPZ8lk4HHZ](https://user-images.githubusercontent.com/78237982/117063563-a1189680-acea-11eb-94d5-a019fd527bcc.png)

**Comment by hadynz** [comment]
You are not comparing apples to apples.

The plugin uses your `My Clippings.txt` file, and the data in that file only. I believe the screenshots you are looking at are from another application (Calibre?). 

If you open the `My Clippings.txt` file that you sent me by email, you will see that "BBC Mundo (calibre)" only has 5 highlights, which is what the plugin has produced.

From what I can see so far, the plugin is parsing the data in your file as expected. Is it possible that you sent me an outdated `My Clippings.txt` file that doesn't have all your latest highlights?

![image](https://user-images.githubusercontent.com/315585/117278159-9dd10680-aeb4-11eb-8006-31e0993989e3.png)

**Comment by EhuddR** [comment]
Oh! my bad I thhink I make a mistake whit those files i tried again and the plugin works as expected. 

Thank you very much.
